artemis-commons/src/main/java/org/apache/activemq/artemis/utils/LazyHashProcessor.java

```diff
@@ -0,0 +1,63 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.activemq.artemis.utils;
+
+public abstract class LazyHashProcessor implements HashProcessor {
+   private HashProcessor hashProcessor;
+   private RuntimeException supplierException;
+
+   public HashProcessor getHashProcessor() {
+      if (supplierException != null) {
+         throw supplierException;
+      }
+
+      if (hashProcessor == null) {
+         synchronized (this) {
+            if (supplierException != null) {
+               throw supplierException;
+            }
+
+            if (hashProcessor == null) {
+               try {
+                  hashProcessor = createHashProcessor();
+               } catch (RuntimeException e) {
+                  supplierException = e;
+                  throw supplierException;
+               } catch (Exception e) {
+                  supplierException = new RuntimeException(e);
+                  throw supplierException;
+               }
+            }
+         }
+      }
```

Comment:
This will still be non-thread-safe because the hashProcessor field isnt volatile (prior to the synchronized addition the other one would have needed to be also). 
https://en.wikipedia.org/wiki/Double-checked_locking#Usage_in_Java

Reply:
fixed